### ISSUE TITLE
Make use of `dplyr::mutate()` for `write_dwc()`

### ISSUE THREAD

**Issue by peterdesmet** [inbo/camtraptor#198]
`write_dwc()` spins up a SQLite database to transform Camtrap DP to Darwin Core. This was mainly done to be able to document the transformation in [SQL](https://github.com/inbo/camtraptor/tree/main/inst/sql), **allowing others to make use of the transformation outside of camtraptor**. The main use case I had in mind were pipelines (Agouti, GBIF data ingestion) building DwC-A from Camtrap DP. The downside of this choice is performance and 2 extra dependencies (DBI, RSQLite).

There is now a test implementation of the **GBIF IPT** that supports Camtrap DP as a publication format. This also requires an update to the **GBIF data ingestion** pipeline, but rather than implementing their own transformation, it makes direct use of our `write_dwc()` function (by running camtraptor in a docker container). This results in:

1. A reduced need for users to create Darwin Core Archives (from Camtrap DP) and thus a reduced need to have SQL documentation
2. An increased need for performance in an operational pipeline

We (@PietrH @damianooldoni @peterdesmet) therefore think it would be better to do the Darwin Core transformation with `dplyr::mutate(.keep = "none")` rather than SQL. Since dplyr is inspired by SQL, the code will still be fairly easy to read (while being a lot faster):

```R
occurrence <- mutate(data,
  # DwC term = Source term, variable, static value
  occurrenceID = .data$observationID,
  type = "image",
  license = license,
  rightsHolder = rightsHolder,
  ...
)
```

Ping @fmendezh

**Comment by peterdesmet** [comment]
Note: since this is a non-breaking change I think this can be implemented as part of v0.20. It will require another update once we adopt Camtrap DP 0.6, but it should result in a clearer git history and easier validation if we do this in 2 separate steps (sql -> dplyr, camtrap dp 0.1.6 -> 0.6).

**Comment by damianooldoni** [comment]
I think the same. It will make the changes much more clear to read as extra plus. Thanks to adding it to the [v0.20](https://github.com/inbo/camtraptor/milestone/2) milestone.
